lib/newrelic_httprb/wrappers.rb

```diff
@@ -0,0 +1,51 @@
+module NewRelicHTTP
+  class HTTPResponse
+    attr_reader :response
+
+    def initialize(response)
+      @response = response
+    end
+
+    def [](key)
+      response.headers.each do |k,v|
+        if key.downcase == k.downcase
+          return v
+        end
+      end
+      nil
+    end
+
+    def to_hash
+      response.headers
+    end
+  end
```

Comment:
Good catch. It happens to be the same implementation as the wrapped class responds to the same `#headers` method. I will follow your suggestion to avoid licensing issues.